commands/utils/OutputChannel.ts

```diff
@@ -0,0 +1,58 @@
+import { ContainerRegistryManagementClient } from 'azure-arm-containerregistry/lib/containerRegistryManagementClient';
+import { BlobService, createBlobServiceWithSas } from "azure-storage";
+import * as fs from 'fs';
+import * as os from 'os';
+import * as tar from 'tar';
+import * as url from 'url';
+import vscode = require('vscode');
+import { getBlobInfo } from "../../utils/Azure/acrTools";
+
+const idPrecision = 6;
+const vcsIgnoreList = ['.git', '.gitignore', '.bzr', 'bzrignore', '.hg', '.hgignore', '.svn']
+
+export async function uploadSourceCode(status: vscode.OutputChannel, client: ContainerRegistryManagementClient, registryName: string, resourceGroupName: string, sourceLocation: string, tarFilePath: string, folder: vscode.WorkspaceFolder): Promise<string> {
```

Comment:
>uploadSourceCode [](start = 22, length = 16)

The file name `OutputChannel ` is a little misleading. Maybe you should rename to "sourceArchiveUtility"

Reply:
Done on last commit.